sdk/python/kfp/compiler/_default_transformers.py

```diff
@@ -68,3 +76,20 @@ def _add_pod_labels(task):
         return task
 
     return _add_pod_labels
+
+
+def add_name_for_oob_components() -> Callable:
+    """Adds the OOB component name if applicable."""
+    
+    def _add_name_for_oob_components(task):
+        # Detect the component origin uri in component_ref if exists, and
+        # attach the OOB component name as a pod label.
+        if hasattr(task, '_component_ref'):
+            component_ref = task._component_ref.to_dict()
+            origin_uri = component_ref.get('url')
+            component_name = telemetry.get_component_name(origin_uri)
```

Comment:
Can we use `component_ref.digest` if the URL does not match?

Reply:
Sure we can. For my education: what's the motivation of using digest, is it for caching?

Reply:
Caching is unaffected by this. The digest allows us to measure the usage of all components.
Otherwise we're only measuring a subset.
//For example, Gus has asked for usage stats about the TFX components that we've recently deprecated.